idaes/tests/performance/performance_test_base.py

```diff
@@ -0,0 +1,149 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Author: Andrew Lee
+"""
+import gc
+
+import pyomo.common.unittest as unittest
+from pyomo.util.check_units import assert_units_consistent
+from pyomo.common.timing import TicTocTimer
+from pyomo.environ import assert_optimal_termination
+
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+# Get default solver for testing
+solver = get_solver()
+
+
+class PerformanceBaseClass:  # unittest.TestCase):
+    """
+    Base class for running the IDAES Performance Test Suite.
+
+    This class is intended to be used as a basis for constructing tests to be run as part of the IDAES
+    performance testing suite, and should implement the following methods:
+
+    * build_model - should return a complete instance of the model being tested.
+    * test_performance - a test method for profiling the model of interest which logs some performance data.
+        A standard method is provided which logs some common performance data, but developers should overload this
+        as required.
+    * initialize_model - (optional) a method to initialize the model of interest. This method is called as part
+        of the standard test_performance method.
+    * solve_model - (optional) a method to solve the model of interest. This method is called as part
+        of the standard test_performance method, and a standard method is provided by the base class.
+
+    Developers may add additional test methods to the derived class to run additional performance tests if desired.
+    These methods should ass start with 'test_' in order ot be discovered by pytest.
+
+    Usage:
+
+    Performance tests should inherit from this based class and decorate the derived class with the 'performance'
+    pytest mark. This will automatically categorize all tests in this class as performance tests which will only be run
```

Comment:
Will these line lengths be shortened when `black` is run?

Reply:
Probably not - they are either within the limit we are setting for `black` or are ignored (my pre-commit hook would have already fixed this).